code/crates/starknet/host/src/streaming.rs

```diff
@@ -8,63 +10,69 @@ use malachitebft_core_types::Round;
 use malachitebft_engine::util::streaming::{Sequence, StreamId, StreamMessage};
 
 use crate::types::{Address, Height, ProposalInit, ProposalPart};
+use std::fmt::Debug;
 
-struct MinSeq<T>(StreamMessage<T>);
+// Wraps StreamMessage to implement custom ordering for a BinaryHeap
+// The default BinaryHeap is a max-heap, so we reverse the ordering
+// by implementing Ord in reverse to make it min-heap, which suits the purpose of efficiently
+// providing available proposal part with smallest sequence number
+#[derive_where(Debug)]
+pub struct MinSeq<T: Debug>(pub StreamMessage<T>);
```

Comment:
No need to include an explicit bound on `T`, instead you can do:

```suggestion
#[derive(Debug)]
pub struct MinSeq<T>(pub StreamMessage<T>);
```

and remove all the `Debug` bounds below.